qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java

```diff
@@ -0,0 +1,581 @@
+
+  @SuppressWarnings("unused")
+  @RegisterExtension
+  final ContainerLogsDumper logsWatcher =
+      new ContainerLogsDumper(
+          () -> Map.of("postgres", POSTGRES, "keycloak", KEYCLOAK, "identity", IDENTITY));
+
+  private BpmnModelInstance process;
+
+  @BeforeAll
+  static void init() throws Exception {
+    ZEEBE
+        .withGatewayConfig(
+            gateway -> {
+              gateway.getMultiTenancy().setEnabled(true);
+              gateway.getSecurity().getAuthentication().setMode(AuthMode.IDENTITY);
+              gateway
+                  .getSecurity()
+                  .getAuthentication()
+                  .getIdentity()
+                  .setBaseUrl(
+                      "http://%s:%d".formatted(IDENTITY.getHost(), IDENTITY.getMappedPort(8080)));
+              gateway
+                  .getSecurity()
+                  .getAuthentication()
+                  .getIdentity()
+                  .setIssuerBackendUrl(
+                      "http://%s:%d%s"
+                          .formatted(
+                              KEYCLOAK.getHost(),
+                              KEYCLOAK.getMappedPort(8080),
+                              KEYCLOAK_PATH_CAMUNDA_REALM));
+              gateway
+                  .getSecurity()
+                  .getAuthentication()
+                  .getIdentity()
+                  .setAudience(ZEEBE_CLIENT_AUDIENCE);
+            })
+        .start()
+        .await(TestHealthProbe.READY);
+
+    awaitCamundaRealmAvailabilityOnKeycloak();
+
+    associateTenantsWithClient(List.of("tenant-a"), ZEEBE_CLIENT_ID_TENANT_A);
+    associateTenantsWithClient(List.of("tenant-b"), ZEEBE_CLIENT_ID_TENANT_B);
+    associateTenantsWithClient(List.of("tenant-a", "tenant-b"), ZEEBE_CLIENT_ID_TENANT_A_AND_B);
+  }
+
+  @BeforeEach
+  void setup() {
+    process =
+        Bpmn.createExecutableProcess(Strings.newRandomValidBpmnId())
+            .startEvent()
+            .serviceTask("task", b -> b.zeebeJobType("type"))
+            .endEvent()
+            .done();
+  }
+
+  @Test
+  void shouldDenyDeployProcessWhenUnauthorized() {
+    // given
+    try (final var client = createZeebeClient(ZEEBE_CLIENT_ID_TENANT_A)) {
+      // when
+      final Future<DeploymentEvent> result =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(process, "process.bpmn")
+              .tenantId("tenant-b")
+              .send();
+
+      // then
+      assertThat(result)
+          .failsWithin(Duration.ofSeconds(10))
+          .withThrowableThat()
+          .withMessageContaining("PERMISSION_DENIED")
+          .withMessageContaining(
+              "Expected to handle gRPC request DeployResource with tenant identifier `tenant-b`")
+          .withMessageContaining("but tenant is not authorized to perform this request");
+    }
+  }
+
+  @Test
+  void shouldAuthorizeCreateProcessInstance() {
```

Comment:
:wrench: It might be useful to attempt the creation of a process instance with a call activity, where the called process also belongs to the same tenant, or another case where the called process belongs to a different tenant (so a failure is expected).